Traslate YDB concepts (#8172)
ydb-platform/ydb: ydb/docs/en/core/concepts/_includes/index/how_it_works.md

One of the key building blocks of {{ ydb-short-name }}'s compute layer is called a *tablet*. They are stateful logical components implementing various aspects of {{ ydb-short-name }}.

The next level of detail of overall {{ ydb-short-name }} architecture is explained in the [YDB cluster documentation section](../../cluster/index.md).

### Hierarchy {#ydb-hierarchy}

![Hierarchy](https://storage.yandexcloud.net/ydb-www-prod-site-assets/howitworks/organization.png)

From the user's perspective, everything inside {{ ydb-short-name }} is organized in a hierarchical structure using directories. It can have arbitrary depth depending on how you choose to organize your data and projects. Even though {{ ydb-short-name }} does not have a fixed hierarchy depth like in other SQL implementations, it will still feel familiar as this is exactly how any virtual filesystem looks like.

### Table {#table}

![Table](https://storage.yandexcloud.net/ydb-www-prod-site-assets/howitworks/table.png)

> {{ ydb-short-name }} provides users with the well-known abstraction — tables. In {{ ydb-short-name }}, there are two main types of tables:
> * [Row tables](../../datamodel/table.md#row-tables) are designed for OLTP workloads.
> * [Column tables](../../datamodel/table.md#column-tables) are designed for OLAP workloads.

Reviewer: ```suggestion
{{ ydb-short-name }} provides users with a well-known abstraction — tables. In {{ ydb-short-name }}, there are two main types of tables:

* [Row-oriented tables](../../datamodel/table.md#row-tables) are designed for OLTP workloads.
* [Column-oriented tables](../../datamodel/table.md#column-tables) are designed for OLAP workloads.
```

> Logically, from the user's point of view, both types of tables look the same. The main difference between row tables and column tables lies in the way data is stored. In row tables, the values of all columns in each row are stored together, while in column tables — on the contrary, each column is stored separately, so cells related to different rows are stored next to each other.
> 
> Regardless of its type, each table must contain a primary key. Column tables can use only  `NOT NULL` columns as primary keys. Data in tables is physically sorted by the primary key. Partitioning works differently in row and column tables:
> * Row tables are automatically partitioned by the ranges of primary keys depending on the volume of data.
> * Column tables are partitioned by the hash of the partitioning columns.
> 
> Each range of primary keys in a table is processed by a specific [tablet](../../cluster/common_scheme_ydb.md#tablets), called a data shard for row tables and a column shard for column tables.

Reviewer: ```suggestion
Logically, from the user's perspective, both types of tables look the same. The main difference between row-oriented tables and column-oriented tables is in how the data is physically stored. In row-oriented tables, the values of all columns in each row are stored together. In contrast, in column-oriented tables, each column is stored separately, meaning cells related to different rows are stored next to each other.

Regardless of type, each table must have a primary key. Column tables can only use `NOT NULL` columns as primary keys. Table data is physically sorted by the primary key. 

Partitioning works differently in row and column tables:

* Row-oriented tables are automatically partitioned by primary key ranges, depending on the data volume.
* Column-oriented tables are partitioned by the hash of the partitioning columns.

Each partition of a table is processed by a specific [tablet](../../glossary.md#tablets), called a [data shard(../../glossary.md#datashard) for row-oriented tables and a [column shard](../../glossary.md#columnshard) for column-oriented tables.
```

The next level of detail of overall {{ ydb-short-name }} architecture is explained in the [YDB cluster documentation section](../../cluster/index.md).

### Hierarchy {#ydb-hierarchy}

![Hierarchy](https://storage.yandexcloud.net/ydb-www-prod-site-assets/howitworks/organization.png)

From the user's perspective, everything inside {{ ydb-short-name }} is organized in a hierarchical structure using directories. It can have arbitrary depth depending on how you choose to organize your data and projects. Even though {{ ydb-short-name }} does not have a fixed hierarchy depth like in other SQL implementations, it will still feel familiar as this is exactly how any virtual filesystem looks like.

### Table {#table}

![Table](https://storage.yandexcloud.net/ydb-www-prod-site-assets/howitworks/table.png)

{{ ydb-short-name }} provides users with a well-known abstraction — tables. In {{ ydb-short-name }}, there are two main types of tables:

* [Row-oriented tables](../../datamodel/table.md#row-tables) are designed for OLTP workloads.
* [Column-oriented tables](../../datamodel/table.md#column-tables) are designed for OLAP workloads.

> Logically, from the user's perspective, both types of tables look the same. The main difference between row-oriented tables and column-oriented tables is in how the data is physically stored. In row-oriented tables, the values of all columns in each row are stored together. In contrast, in column-oriented tables, each column is stored separately, meaning cells related to different rows are stored next to each other.
> 
> Regardless of type, each table must have a primary key. Column tables can only use `NOT NULL` columns as primary keys. Table data is physically sorted by the primary key.

Reviewer: ```suggestion
Logically, from the user’s perspective, both types of tables look the same. The main difference between row-oriented and column-oriented tables lies in how the data is physically stored. In row-oriented tables, the values of all columns in each row are stored together. In contrast, in column-oriented tables, each column is stored separately, meaning that cells from different rows are stored next to each other within the same column.

Regardless of the type, each table must have a primary key. Column-oriented tables can only have `NOT NULL` columns in primary keys. Table data is physically sorted by the primary key.
```

> Partitioning works differently in row and column tables:
> 
> * Row-oriented tables are automatically partitioned by primary key ranges, depending on the data volume.
> * Column-oriented tables are partitioned by the hash of the partitioning columns.
> 
> Each partition of a table is processed by a specific [tablet](../../glossary.md#tablets), called a [data shard(../../glossary.md#datashard) for row-oriented tables and a [column shard](../../glossary.md#columnshard) for column-oriented tables.

Reviewer: ```suggestion
Partitioning works differently in row-oriented and column-oriented tables:

* Row-oriented tables are automatically partitioned by primary key ranges, depending on the data volume.
* Column-oriented tables are partitioned by the hash of the partitioning columns.

Each partition of a table is processed by a specific [tablet](../../glossary.md#tablets), called a [data shard](../../glossary.md#datashard) for row-oriented tables and a [column shard](../../glossary.md#columnshard) for column-oriented tables.
```
